### PR TITLE
function to change the terraform statefile

### PR DESCRIPTION
**Pull request by vavuthu**
function to change the terraform statefile which is needed for node replacement

Signed-off-by: vavuthu <vavuthu@redhat.com>

### CODE REVIEW

**Comment by vavuthu** [comment]
verified locally and teardown as well

logs:
```
-------------------------------------------------------------------------------------------------- live log call --------------------------------------------------------------------------------------------------
11:17:52 - MainThread - ocs_ci.deployment.terraform - INFO  - modifying terraform state file
11:17:52 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: terraform state rm 'module.compute_vm.vsphere_virtual_machine.vm["compute-0.j-056vu1cs33l-a.qe.rh-ocs.com"]'
PASSED
```
teradown
```
11:27:48 - MainThread - ocs_ci.deployment.terraform - INFO  - Initializing terraform work directory
11:27:48 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: /home/vavuthu/VJ/projects/ocs-ci/bin/terraform init /home/vavuthu/VJ/projects/ocs-ci/external/installer/upi/vsphere/
11:27:58 - MainThread - ocs_ci.deployment.terraform - INFO  - Destroying the cluster
11:27:58 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: /home/vavuthu/VJ/projects/ocs-ci/bin/terraform destroy -refresh=false '-var-file=/home/vavuthu/VJ/clusterdirs/j-056vu1cs33l-a/terraform_data/terraform.tfvars' -auto-approve /home/vavuthu/VJ/projects/ocs-ci/external/installer/upi/vsphere/
p11:38:53 - MainThread - ocs_ci.deployment.vmware - INFO  - Resource pool j-056vu1cs33l-a does not exist in cluster Cluster-2

```

File: ocs_ci/deployment/terraform.py [hunk 1]
```diff
@@ -121,3 +121,24 @@ def destroy_module(self, tfvars, module):
...
         cmd = f"terraform destroy -auto-approve -target={module} '-var-file={tfvars}' '{self.path}'"
         run_cmd(cmd, timeout=1200)
+
+    def change_statefile(self, module, resource_type, resource_name, instance):
+        """
+        Remove the records from the state file so terraform will no longer be
+        tracking the corresponding remote objects.
+
+        Args:
+            tfstate (str): path to terraform.tfstate file
```
Comment (yitzhak12) [inline ocs_ci/deployment/terraform.py:131]: This param is not in use in the function.

File: ocs_ci/deployment/terraform.py [hunk 2]
```diff
@@ -121,3 +121,24 @@ def destroy_module(self, tfvars, module):
...
+            module (str): Name of the module
+                e.g: compute_vm, module.control_plane_vm etc.
+            resource_type (str): Resource type
+                e.g: vsphere_virtual_machine, vsphere_compute_cluster etc.
+            resource_name (str): Name of the resource
+                e.g: vm
+            instance (str): Name of the instance
+                e.g: compute-0.j-056vu1cs33l-a.qe.rh-ocs.com
```
Comment (yitzhak12) [inline ocs_ci/deployment/terraform.py:139]: Should I pass the name `compute-0` or the full name `compute-0.j-056vu1cs33l-a.qe.rh-ocs.com`? How can I retrieve the name above?

File: ocs_ci/deployment/terraform.py (hunk 1 above)
Comment (vavuthu) [reply ocs_ci/deployment/terraform.py]: done

**Comment by yitzhak12** [comment]
@vavuthu Can you provide an example of the usage of the function above? For example, what arguments should be after deleting the worker node `compute-1`?

File: ocs_ci/deployment/terraform.py (hunk 2 above)
Comment (vavuthu) [reply ocs_ci/deployment/terraform.py:138]: you have to pass the full name. one can form the full name with below

f"compute-0.{config.ENV_DATA.get('cluster_name')}.{config.ENV_DATA.get('base_domain')}"

Comment (yitzhak12) [reply ocs_ci/deployment/terraform.py:138]: Okay, I see.

**Review by yitzhak12 — DISMISSED** [review]

**Review by petr-balogh — APPROVED** [review]

**Review by yitzhak12 — APPROVED** [review]

**Pull request merged**